library/src/main/java/me/grantland/widget/AutofitHelper.java

```diff
@@ -128,19 +131,49 @@ private static void autofit(TextView view, TextPaint paint, float minTextSize, f
                     displayMetrics);
         }
 
-        if (size < minTextSize) {
-            size = minTextSize;
+        if (mIsHeightFitting) {
+            int targetHeight = view.getHeight() - view.getPaddingTop() - view.getPaddingBottom();
+            if (targetHeight <= 0) {
+                if (size < minTextSize) {
+                    size = minTextSize;
+                }
+                view.setTextSize(TypedValue.COMPLEX_UNIT_PX, size);
+                return;
+            }
+
+            float textHeight = getTextHeight(text, paint, targetWidth, size);
+            float heightRatio = targetHeight / textHeight;
+            float newSize = size * heightRatio;
+            if (newSize < size) {
+                size = newSize;
+            }
+
+            if (size < minTextSize) {
+                size = minTextSize;
+            }
```

Comment:
I think we need to pop this conditional out of it's nested conditional or else we'll break minTextSize for non-height-fitting modes